src/nginxconfig/templates/domain_sections/rails.vue

```diff
@@ -0,0 +1,167 @@
+                            </span>
+                        </label>
+                    </div>
+                </div>
+            </div>
+        </div>
+
+    </div>
+</template>
+
+<script>
+    import PrettyCheck from 'pretty-checkbox-vue/check';
+    import delegatedFromDefaults from '../../util/delegated_from_defaults';
+    import computedFromDefaults from '../../util/computed_from_defaults';
+
+    const defaults = {
+        rails: {
+            default: false,
+            enabled: false,
+        },
+        namedLocation: {
+            default: '',
+            placeholder: '@my_awesome_rails_app',
+            enabled: true,
+        },
+    };
+
+    export default {
+        name: 'DomainRails',                               // Component name
+        display: 'common.rails',                           // Display name for tab (i18n key)
+        key: 'rails',                                      // Key for data in parent
+        delegated: delegatedFromDefaults(defaults),         // Data the parent will present here
+        components: {
+            PrettyCheck,
+        },
+        props: {
+            data: Object,                                   // Data delegated back to us from parent
+        },
+        computed: {
+            ...computedFromDefaults(defaults, 'rails'), // Getters & setters for the delegated data
+            incorrectBeginning() {
+                return this.namedLocationChanged && !this.$props.data.namedLocation.computed.startsWith('@');
+            },
+            hasWarnings() {
+                return this.incorrectBeginning;
+            },
+        },
+        watch: {
+            // If the Reverse proxy or PHP is enabled, rails will be forced off
+            '$parent.$props.data': {
+                handler(data) {
+                    // This might cause recursion, but seems not to
+                    if (data.reverseProxy.reverseProxy.computed || data.php.php.computed) {
+                        this.$props.data.rails.enabled = false;
+                        this.$props.data.rails.computed = false;
+                    } else {
+                        this.$props.data.rails.enabled = true;
+                        this.$props.data.rails.computed = this.$props.data.rails.value;
+                    }
+                },
+                deep: true,
+            },
+            // Set Rails namedLocation
+            '$props.data.namedLocation': {
+                handler(data) {
+                    if (data.computed) {
+                        console.log(this.$props.data.namedLocation.value);
+                        data.computed = this.$props.data.namedLocation.value;
+                        return data.computed;
+                    } else {
+                        data.computed = false;
+                    }
+                },
+                deep: true,
+            },
```

Comment:
Is this listener needed? Does computed need to be forced to `false` if the string is empty? The default value is an empty string, not `false`?